protocol/0092-SAMM-spot_automated_market_maker.md

```diff
@@ -0,0 +1,223 @@
+
+For a given price $p$ calculate implied position $P_i$ with
+
+$$
+P_i = L \cdot \frac{\sqrt{p_u} - \sqrt{p}}{\sqrt{p} \cdotp \sqrt{p_u}} ,
+$$
+
+Then simply return the absolute difference between these two prices.
+
+## Determining Liquidity Contribution
+
+Liquidity contribution for spot AMMs should be determined identically to that for futures market vAMMs in [0089-SAMM](./0089-SAMM-automated_market_maker.md)
+
+## Setting Fees
+
+The `proposed_fee` provided as part of the AMM construction contributes to the fee determination logic on the market, if a setup where LPs decide on the market fee is in use. In the case where it is the AMM's current assigned ELS, or the running average liquidity provided so far if the commitment was made in the current epoch, is used for weighting the AMM's vote for the fee.
+
+## Market Settlement
+
+At market settlement, an AMM's position will be settled alongside all others as if they are a standard party. Once settlement is complete, any remaining funds in the AMM's account will be transferred back to the creator's general account and the AMM can be removed.
+
+
+## Acceptance Criteria
+
+- When `market.amm.minCommitmentQuantum` is `1`, mid price of the market `ETH/USDT` is `100`, a user with `1000 USDT` is able to create an AMM with commitment `1000 USDT`, lower price `100`, upper price `150`, reference price `100`. (<a name="0092-SAMM-001" href="#0092-SAMM-001">0092-SAMM-001</a>)
+- When `market.amm.minCommitmentQuantum` is `1`, mid price of the market `ETH/USDT` is `100`, a user with `1000 USDT` is unable to create an AMM with commitment `1000 USDT`, lower price `100`, upper price `150`, reference price `150`. (<a name="0092-SAMM-002" href="#0092-SAMM-002">0092-SAMM-002</a>)
+- When `market.amm.minCommitmentQuantum` is `1`, mid price of the market `ETH/USDT` is `100`, a user with `1000 USDT` is able to create an AMM with commitment `1 ETH`, lower price `100`, upper price `150`, reference price `150`. (<a name="0092-SAMM-003" href="#0092-SAMM-003">0092-SAMM-003</a>)
+- When `market.amm.minCommitmentQuantum` is `100`, mid price of the market `ETH/USDT` is `100`, a user with `10 USDT` is unable to create an AMM with commitment `10 USDT`, lower price `100`, upper price `150`, reference price `150`. (<a name="0092-SAMM-004" href="#0092-SAMM-004">0092-SAMM-004</a>)
+
+- When `market.amm.minCommitmentQuantum` is `1`, mid price of the market `ETH/USDT` is `100`, a user with `1 ETH` is able to create an AMM with commitment `1 ETH`, lower price `80`, upper price `100`, reference price `100`. (<a name="0092-SAMM-005" href="#0092-SAMM-005">0092-SAMM-005</a>)
+- When `market.amm.minCommitmentQuantum` is `1`, mid price of the market `ETH/USDT` is `100`, a user with `1 ETH` is unable to create an AMM with commitment `1 ETH`, lower price `80`, upper price `100`, reference price `80`. (<a name="0092-SAMM-006" href="#0092-SAMM-006">0092-SAMM-006</a>)
```

Comment:
Additional AC covering a case where user has both of the assets might be useful.